reana_client/validation.py

```diff
@@ -521,6 +521,39 @@ def _validate_cwl_parameters(reana_yaml):
         )
         sys.exit(1)
 
+    dang_params = _validate_cwl_dangerous_operations(reana_yaml)
```

Comment:
We already have [`_validate_dangerous_operations`](https://github.com/reanahub/reana-client/blob/20dc10301b74766b3f9c27daba6608d1e65a3840/reana_client/validation.py#L640-L654), please use it to have consistent warning messages for all the workflow engines.

I see that for CWL the potential dangerous operations can be both in `baseCommand` and `arguments`. Some examples:
- https://github.com/reanahub/reana-demo-cdci-crab-pulsar-integral-verification/blob/2674d484cd2cb1ac8c00068e6bc53804be31b999/workflow/cwl/crab.cwl#L3-L7
- https://github.com/reanahub/reana-demo-root6-roofit/blob/a43db4aad1c3e1e2c3f7eb8d352169c0eba1beb6/workflow/cwl/fitdata.cwl#L20-L26

I wonder if we should also check `inputs` as they may contain some bash statements inside.. 🤔, although this wouldn't be a good practice at all so we can pass, it's users' resposibility.

So the idea would be simply passing the value of those fields to `_validate_dangerous_operations`, which will take care of doing the actual checks.

Reply:
I am now using the `_validate_dangerous_operations` and warning are shown like as below. 

<img width="957" alt="Screenshot 2021-03-18 at 12 14 50 PM" src="https://user-images.githubusercontent.com/24358501/111617551-8e520d00-87e3-11eb-977f-430499080112.png">

Reply:
Judging from your screenshot. `arguments` and `baseCommand` are not steps, steps are e.g. `gendata` and `fitdata` from `r-d-root6-roofit` example.

Reply:
okay, Should I name them as `param` or `Operation "cd /" found in "arguments" might be dangerous` is enough?

Reply:
No, it's not enough. The idea here is that if the user has a workflow with tens of steps, is able to quickly identify where this dangerous operation is. Please check the output when introducing a dangerous operation for Serial or Yadage workflows in `r-d-root6-roofit`:

```console
$ reana-client validate -f reana-yadage.yaml               
==> Verifing REANA specification file... /Users/marco/code/reanahub/reana-demo-root6-roofit/reana-yadage.yaml
  -> SUCCESS: Valid REANA specification file.
==> Verifing workflow parameters and commands... 
  -> WARNING: Operation "cd /" found in step "gendata" might be dangerous.
  -> WARNING: Operation "sudo" found in step "fitdata" might be dangerous.
```

The output should be exactly the same for CWL.

Take into account that a step might have no name, (e.g. `r-d-worldpopulation`). In that case, we show the number.

```console
$ reana-client validate                     
==> Verifing REANA specification file... /Users/marco/code/reanahub/reana-demo-worldpopulation/reana.yaml
  -> SUCCESS: Valid REANA specification file.
==> Verifing workflow parameters and commands... 
  -> WARNING: Operation "sudo" found in step "0" might be dangerous.
```

Reply:
The difference in CWL workflows is these `arguments` and `baseCommand` may/may not be be included in the steps. In `reana-demo-cdci-crab-pulsar-integral-verification` it is present in `workflow spec` directly not in any step.

Reply:
I see.. it seems that the cdci example is a bit special as it doesn't have a `Worlkflow` file but only a `CommandLineTool` one, which by checking the docs:

```
A CommandLineTool is a process characterized by the execution of a standalone, non-interactive program which is invoked on some input, produces output, and then terminates.
```

So it looks like a basic execution, no steps involved, which seems a bit weird to me.

So what we can do is making `step` optional in `_validate_dangerous_operations`, `None` by default. If `None`, we can omit the "found in step {}" part. So we can check if the workflow has steps, and pass the proper values. Please a a new test case for this in [test_validate_dangerous_operations](https://github.com/reanahub/reana-client/blob/20dc10301b74766b3f9c27daba6608d1e65a3840/tests/test_validate_parameters.py#L243).

Reply:
As we see that this behavior is limited to only `cwl` workflow. what about sending `step=None` for CWL and keep as it is for other?

Reply:
@mvidalgarcia  
<img width="979" alt="Screenshot 2021-03-18 at 3 26 08 PM" src="https://user-images.githubusercontent.com/24358501/111642313-455b8200-87fe-11eb-9a68-2790abeec0fb.png">
How does it look now?

Reply:
> As we see that this behavior is limited to only `cwl` workflow. what about sending `step_flag=None` for CWL and keep as it is for other?

I don't see the benefit of adding a flag. The step itself can serve as a flag. If it's passed it means we have to display it on the message, if it's not passed, we don't show it.